DMCompiler/DMStandard/_Standard.dm

```diff
@@ -327,19 +327,21 @@ proc/get_dir(atom/Loc1, atom/Loc2)
 			if (Angle != 0)
 				return pick(NORTH, SOUTH, EAST, WEST)
```

Comment:
An invalid dir with non-zero angle can also return random diagonal directions.
An invalid dir with a zero angle will return that same dir, unless it isn't a number in which case it's an exception.
```suggestion
			if (Angle != 0)
				return pick(NORTH, SOUTH, EAST, WEST, NORTHEAST, SOUTHEAST, SOUTHWEST, NORTHWEST)
			else if (!isnum(Dir))
				CRASH("Invalid Dir \"[json_encode(Dir)]\"")
			else
				return Dir
```